bigquery/indexer.py

```diff
@@ -172,11 +111,11 @@ def main():
     args = parse_args()
 
     json_path = os.path.join(args.dataset_config_dir, 'dataset.json')
-    dataset_config = open_and_return_json(json_path)
-    index_name = convert_to_index_name(dataset_config['name'])
+    dataset_config = indexer_util.open_and_return_json(json_path)
+    index_name = indexer_util.convert_to_index_name(dataset_config['name'])
     primary_key = dataset_config['primary_key']
 
-    es = init_elasticsearch(args.elasticsearch_url, index_name)
+    es = indexer_util.init_elasticsearch(args.elasticsearch_url, index_name)
```

Comment:
Since this sequence of calls is always needed to use the utils library properly, change the signature of init_elasticsearch to `(url, json_path)` and have it call `open_and_return_json` and `convert_to_index_name` privately.

Reply:
dataset_config is needed in this file. We get index_name and primary_key from dataset_config. Both index_name and primary_key are used later on.

Reply:
To clarify, `init_elasticsearch` would return `(es, index_name, primary_key)`

Reply:
primary_key is relevant to bigquery indexer, but not necessarily other indexers

Reply:
In that case, we should document in `init_elasticsearch` the specifications for what constitutes a valid `index_name`

Reply:
Done